### PR TITLE
harmonize compliance profiles view with supermarket views

### PR DESCRIPTION
**Pull request by chris-rock**
This PR:
- updates the documentation
- updates the `inspec compliance profiles` list to our supermarket view
- allows `compliance://admin/profile` and `admin/profile` as input
screen shot 2017-04-12 at 11 04 53 pm

### CODE REVIEW

File: lib/bundles/inspec-compliance/cli.rb [hunk 1]
```diff
@@ -57,7 +57,14 @@ def login(server) # rubocop:disable Metrics/AbcSize
       puts '', msg
     end
 
-    desc "login_automate SERVER --user='USER' --ent='ENT' --dctoken or --usertoken='TOKEN'", 'Log in to an Automate SERVER'
+    desc "login_automate SERVER --insecure --user='USER' --ent='ENT' --usertoken='TOKEN'", 'Log in to an Automate SERVER'
```
Comment (adamleff) [inline lib/bundles/inspec-compliance/cli.rb:60]: I would love to see this be `inspec compliance login --automate` in the future than require a completely separate command. It feels weird to me from a CLI UX perspective. Not a blocker, obviously :)

File: lib/bundles/inspec-compliance/api.rb [hunk 2]
```diff
@@ -22,16 +22,14 @@ def self.profiles(config)
...
-            { org: ps['owner_id'], name: owner }
-          end.flatten
+          mapped_profiles = profiles.values.to_a.flatten
         else
-          mapped_profiles = profiles.map do |owner, ps|
-            ps.keys.map do |name|
-              { org: owner, name: name }
-            end
-          end.flatten
+          mapped_profiles = []
```
Comment (adamleff) [inline lib/bundles/inspec-compliance/api.rb:27]: We could use another Rubyism I love... `#each_with_object`:

```
mapped_profiles = profiles.values.each_with_object([]) do |org, memo|
  memo << org.values
end
```

Again, not a blocker, just passing along something I always try and address when I see an empty object being created just to populate it with enumeration.

File: lib/bundles/inspec-compliance/cli.rb [hunk 3]
```diff
@@ -260,6 +269,16 @@ def logout
 
     private
 
+    # returns a parsed url for `admin/profile` or `compliance://admin/profile`
+    def sanitize_profile_name(profile)
+      if profile.is_a?(String) && URI(profile).scheme == 'compliance'
```
Comment (adamleff) [inline lib/bundles/inspec-compliance/cli.rb]: When would `profile` not be a `String`?

**Review by adamleff — APPROVED** [review]
LGTM. Just a question I'd love an answer to, and some suggestions for future. Nothing blocking except the failing travis test.

**Comment by adamleff** [comment]
@chris-rock Can we fix the failing functional test before merge?

File: lib/bundles/inspec-compliance/cli.rb (hunk 1 above)
Comment (chris-rock) [reply lib/bundles/inspec-compliance/cli.rb:60]: Yes!

File: lib/bundles/inspec-compliance/cli.rb (hunk 3 above)
Comment (chris-rock) [reply lib/bundles/inspec-compliance/cli.rb]: good point :-)

**Pull request merged**